README.md

```diff
@@ -34,7 +42,110 @@ Or install it yourself as:
 
     $ gem install transproc
 
-## Usage
+## Basics
+
+The simple transformations are defined as easy as
+
+```ruby
+increament = Transproc::Function.new(-> (data) { data + 1 })
+increament[1] # => 2
+```
+
+It's easy to compose transformations
+
+```ruby
+to_string = Transproc::Function.new(:to_s.to_proc)
+(increament >> to_string)[1] => '2'
+```
+
+It's easy to use extra arguments to transformation
+
+```ruby
+append = Transproc::Function.new(-> (value, suffix) { value + suffix })
+append_bar = append.with('_bar')
+append_bar['foo'] # => foo_bar
+```
+
+Or even accept other transformation as argument
+
+```ruby
+map_array = Transproc::Function.new(-> (array, fn) { array.map(&fn) })
+map_array.with(to_string).call([1, 2, 3]) # => ['1', '2', '3']
+```
+
+To improve this low-level definition, you can use class methods
+with `Transproc::Registry`
+
+```ruby
+M = Module.new do
+  extend Transproc::Registry
+
+  def self.to_string(value)
+    value.to_s
+  end
+
+  def self.map_array(array, fn)
+    array.map(&fn)
+  end
+end
+M[:map_array, M[:to_string]].([1, 2, 3]) # => ['1', '2', '3']
+```
+
+### Built-in transformations
+
+`transproc` comes with a lot of built-in functions. They come in form of module
+with class methods, which you can import into registry.
+
+* [Coercions](http://www.rubydoc.info/gems/transproc/Transproc/Coercions)
+* [Array transformations](http://www.rubydoc.info/gems/transproc/Transproc/ArrayTransformations)
+* [Hash transformations](http://www.rubydoc.info/gems/transproc/Transproc/HashTransformations)
+* [Class transformations](http://www.rubydoc.info/gems/transproc/Transproc/ClassTransformations)
+* [Proc transformations](http://www.rubydoc.info/gems/transproc/Transproc/ProcTransformations)
+* [Conditional](http://www.rubydoc.info/gems/transproc/Transproc/Conditional)
+* [Recursion](http://www.rubydoc.info/gems/transproc/Transproc/Recursion)
+
+Load everything with
```

Comment:
Think this could be updated to:
----------------------
You can import everything with:

```ruby
module T
  extend Transproc::Registry

  import Transproc::Coercions
  import Transproc::ArrayTransformations
  import Transproc::HashTransformations
  import Transproc::ClassTransformations
  import Transproc::ProcTransformations
  import Transproc::Conditional
  import Transproc::Recursion
end
T[:to_string].call(:abc) # => 'abc'
```

Or import selectively with:

```ruby
module T
  extend Transproc::Registry

  import :to_string, from: Transproc::Coercions, as: :stringify
end
T[:stringify].call(:abc) # => 'abc'
T[:to_string].call(:abc)
# => Transproc::FunctionNotFoundError: No registered function T[:to_string]
```